client/state.go

```diff
@@ -0,0 +1,164 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+
+	v1 "github.com/dapr/go-sdk/dapr/proto/common/v1"
+	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
+	"github.com/pkg/errors"
+)
+
+var (
+	// StateOptionConsistencyDefault is strong
+	StateOptionConsistencyDefault = v1.StateOptions_CONSISTENCY_STRONG
+
+	// StateOptionConcurrencyDefault is last write
+	StateOptionConcurrencyDefault = v1.StateOptions_CONCURRENCY_LAST_WRITE
+
+	// StateOptionRetryPolicyDefault is threshold 3
+	StateOptionRetryPolicyDefault = &v1.StateRetryPolicy{
+		Threshold: 3,
+	}
+
+	// StateOptionDefault is the optimistic state option (last write concurency and strong consistency)
+	StateOptionDefault = &v1.StateOptions{
+		Concurrency: StateOptionConcurrencyDefault,
+		Consistency: StateOptionConsistencyDefault,
+		RetryPolicy: StateOptionRetryPolicyDefault,
+	}
+)
+
+// *** Save State ***
+
+// SaveState saves the fully loaded save state request
+func (c *Client) SaveState(ctx context.Context, req *pb.SaveStateRequest) error {
+	if req == nil {
+		return errors.New("nil request")
+	}
+
+	_, err := c.protoClient.SaveState(authContext(ctx), req)
+	if err != nil {
+		return errors.Wrap(err, "error saving state")
+	}
+
+	return nil
+}
+
+// SaveStateItem saves a single state item
+func (c *Client) SaveStateItem(ctx context.Context, store string, item *v1.StateItem) error {
```

Comment:
this is providing a mix and match of custom methods and underlying auto generated types. Provide wrapper types in the sdk layer, so that auto generated sdk is not bleeding through it.

Reply:
done